random-facts/homogeneity.tex

```diff
@@ -0,0 +1,134 @@
+\begin{definition}
+  A \emph{graded} $R$-algebra $S$ is an $R$-algebra $S$ together with the datum of a direct sum decomposition
+  \begin{equation*}
+    S = \oplus_{n \in \Z} S_n
+  \end{equation*}
+  as an $R$-module such that $S_k \cdot S_\ell \subseteq S_{k + l}$
+  for every $k$, $\ell \in \Z$.  We identify each $S_n$ for $n \in \Z$
+  with its image in $S$.  The elements of $S_n$ are called \emph{homogeneous} of degree $n$.
+\end{definition}
+
+\begin{remark}
+  The datum of a grading of an $R$-algebra $S$ thus gives us an
+  essentially finite decomposition $u = \sum_{n \in \Z} u_n$ for every
+  $u \in S$ where each $u_n$ in homogeneous of degree $n$.
+  Furthermore, the decomposition of a homogeneous element $u$ is just $u = u$.
+\end{remark}
+
+\begin{proposition}
+  \label{Spectrum of graded algebra}
+  Let $S$ be a graded $R$-algebra and let $X \coloneqq \Spec S$.  For each $t \in R^\times$ and each $p \in X$ we define
+  \begin{equation*}
+    t \cdot p \coloneqq \left(u \mapsto \sum_{n \in Z} u_n t^n\right) \in \Spec S.
```

Comment:
Is something missing here? Didn't think long about it, but p does not appear on the right hand side.

Reply:
Good catch.  It has to be u_n(p) on the right-hand side.